### PR TITLE
Support indexes with IDT UMIs

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: scripts/samplesheet_generator.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 DESC = """EPP used to create samplesheets for Illumina sequencing platforms"""
 
 # Pre-compile regexes in global scope:
-IDX_PAT = re.compile("([ATCG]{4,})-?([ATCG]*)")
+IDX_PAT = re.compile("([ATCGN]{5,})-?([ATCG]*)")
```
Comment (jgruselius) [inline scripts/samplesheet_generator.py]: Should work 👍 
Maybe something like `([ATCG]{4,})N*-?([ATCG]*)` if you want to be more specific.

Comment (chuan-wang) [reply scripts/samplesheet_generator.py]: Hi @jgruselius, thank you for the suggestion! I tested and the specific expression should be `([ATCG]{4,}N*)-?([ATCG]*)`. The code has been updated.

Comment (jgruselius) [reply scripts/samplesheet_generator.py]: If you want the N's in the capture group along with index 1, then yes.

**Comment by ewels** [comment]
Nice! Do the UMI sequences that we're masking out here still end up in the FastQ files after this runs? Or are they lost?

File: scripts/samplesheet_generator.py (hunk 1 above)
Comment (ewels) [reply scripts/samplesheet_generator.py]: I don't think that we do want the N's in the capture group do we..?

Comment (chuan-wang) [reply scripts/samplesheet_generator.py]: I guess we need the N's to set correct cycle numbers in the masks?

**Comment by chuan-wang** [comment]
> Nice! Do the UMI sequences that we're masking out here still end up in the FastQ files after this runs? Or are they lost?

They are lost in the last run. The sequencing staffs missed the information of additional indexing cycles.

**Comment by alneberg** [comment]
> > Nice! Do the UMI sequences that we're masking out here still end up in the FastQ files after this runs? Or are they lost?
> 
> They are lost in the last run. The sequencing staffs missed the information of additional indexing cycles.

@chuan-wang, Is this why I wasn't able to fetch them with `bcl2fastq` for [P12151](https://genomics-status.scilifelab.se/project/P12151)? Or is this referring to a different run?

**Comment by chuan-wang** [comment]
@alneberg The error is with another project [P12502](https://genomics-status.scilifelab.se/project/P12502)

**Pull request merged**